sky/backends/cloud_vm_ray_backend.py

```diff
@@ -733,7 +730,7 @@ def _update_blocklist_on_aws_error(
             with ux_utils.print_exception_no_traceback():
                 raise RuntimeError('Errors occurred during provision; '
                                    'check logs above.')
-        if set(zones) == set(region.zones):
+        if region.zones is None or set(zones) == set(region.zones):
```

Comment:
Is the `region.zones is None` clause for Azure and Lambda? If so, wdyt about using a different warning message e.g., "Got error(s) in region {region.name}", since they may not have a zone concept (at least for Lambda)?